### DISCUSSION TITLE
How to refresh the player once received EXT-X-ENDLIST in the manifest

### DISCUSSION THREAD

**Discussion by lunika** [videojs/video.js#7534] [Q&A] [answered]
Hello,

I'm working on a [webinar](https://github.com/openfun/marsha) project and one of the feature is the availability to start and stop again and again the webinar.

The live video is streamed using AWS medialive and mediapackage solution. When a medialive channel is stopped, then the mediapackage channel stop streaming by adding `#EXT-X-ENDLIST` in its last manifest but when we restart streaming, the manifest from MediaPackage becomes live again as explain here https://docs.aws.amazon.com/mediapackage/latest/ug/what-is-manifest.html and videojs player is not starting to play again.

Here an example of the last manifest received:

```
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-TARGETDURATION:4
#EXT-X-MEDIA-SEQUENCE:509
#EXT-X-DISCONTINUITY-SEQUENCE:16
#EXTINF:3.917,
dev-manu_030aaea4-bb0b-4915-88a4-521fc8b59366_1637050264_hls_1_509.ts?m=1637050263
#EXT-X-ENDLIST
```

We are looking for a solution to tell videojs that the streaming is live again and refresh itself, load the manifest again and resume streaming. 

We tried several solution but no one work:

- calling `player.load()`
- calling `player.reset()` then `player.load()`

One solution can work but is not graceful is to dispose the player and recreate one from scratch. Obviously the user will see the player disappear and then appear again like if there is a glitch in the interface.

If someone have an idea to do this gracefully it should be great.

Thanks for your help.

**Comment by gkatsev** [comment, marked as answer]
Does setting the source back in the player not work? For example `player.src(player.currentSource())`.

**Reply by lunika** [reply]
That's a good idea, now I have to deal with the latency to not download the manifest with `#EXT-X-ENDLIST` again in it.

**Reply by lunika** [reply]
It works and it was simple 😅

Thank you
